src/expr/src/relation/join_input_mapper.rs

```diff
@@ -277,37 +277,6 @@ impl JoinInputMapper {
         None
     }
 
-    /// Try to rewrite an subexpression referencing the larger join so that all the
-    /// column references point to the `index` input taking advantage of equivalences
-    /// in the join, if necessary.
-    /// Takes an expression in the global context and makes rewrites in the
-    /// global context so we can identify using `lookup_inputs` whether if an expression
-    /// was only partially rewritten.
-    fn try_map_to_input_with_bound_expr_sub(
-        &self,
-        expr: &mut MirScalarExpr,
-        index: usize,
-        equivalences: &[Vec<MirScalarExpr>],
-    ) {
-        {
-            let mut inputs = self.lookup_inputs(expr);
-            if let Some(first_input) = inputs.next() {
-                if inputs.next().is_none() && first_input == index {
-                    // there is only one input, and it is equal to index, so we're
-                    // good. do not continue the recursion
-                    return;
-                }
-            }
```

Comment:
I took these lines out because I realized that `lookup_inputs` calls `expr.support()`, which then visits all the nodes in the subexpression. Thus, doing this does not save any recursion effort.

Reply:
Never mind. Philip has found a bunch of negative effects from taking out these lines, so I'm putting them back in for the time being.